debian-bo-master/bo/source/misc/dbview-1.0.3/dbview.1
.\" Copyright (c) 1996 Martin Schulze (joey@infodrom.north.de)
.\"
.\" This is free documentation; you can redistribute it and/or
.\" modify it under the terms of the GNU General Public License as
.\" published by the Free Software Foundation; either version 2 of
.\" the License, or (at your option) any later version.
.\"
.\" The GNU General Public License's references to "object code"
.\" and "executables" are to be interpreted as the output of any
.\" document formatting or typesetting system, including
.\" intermediate and printed output.
.\"
.\" This manual is distributed in the hope that it will be useful,
.\" but WITHOUT ANY WARRANTY; without even the implied warranty of
.\" MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
.\" GNU General Public License for more details.
.\"
.\" You should have received a copy of the GNU General Public
.\" License along with this manual; if not, write to the Free
.\" Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139,
.\" USA.
.\"
.TH DBVIEW 1 "October 8, 1996" "Unix" "Database Management"
.SH NAME
dbview \- View dBase III files
.SH SYNOPSIS
.B dbview
.RB [ \-b | \-\-browse ]
.RB [ \-d 
.IR delim |
.B \-\-delimiter
.IR delim ]
.RB [ \-e | \-\-description ]
.RB [ \-h | \-\-help ]
.RB [ \-i | \-\-info ]
.RB [ \-o | \-\-omit ]
.RB [ \-v | \-\-version ]
.RB [ \-r | \-\-reserve ]
.RB [ \-t | \-\-trim ]
.I dbfile
.SH DESCRIPTION
.B Dbview
is a little tool that will display dBase III files.  You can also use
it to convert your old .dbf files for further use with Unix.  It should
also work with dBase IV files, but this is mostly untested.

By default
.B dbview
displays the contents of a dBase III or IV database file.  This is be
done by displaying both the name of the field itself and its value.  At
the end of every record a newline is appended.

.SH OPTIONS
If no option given
.B dbview
only displays the database in its most friendly way.

.TP
.B "\-\-browse, \-b"
switches into browse mode.  Using this mode no fieldnames will be
displayed, instead every record will displayed in one line using a
delimiter to separate fields.

.TP
.BI "\-\-delimiter, \-d "delimiter
The default delimiter in browse mode is the colon sign ``:''.  This
parameter overrides it.  This can be useful especially if you plan to
examine the output with scripts.

.TP
.B "\-\-description, \-e"
displays the field description of the database.

.TP
.B "\-\-help, \-h"
displays a complete (or short) help screen.

.TP
.B "\-\-info, \-i"
displays some (partially technical) information about the database like
number of records and length of each record.

.TP
.B "\-\-omit, \-o"
omits displaying the whole database.  Using this parameter can be
useful if you're only interested in the structure.

.TP
.B "\-\-reserve, \-r"
Normally fieldnames are converted into a more friendly format.  They
are stored in capital letters, but that looks like shouting.  This
parameter supresses the conversion.

.TP
.B "\-\-trim, \-t"
When this option is specified, leading and trailing spaces are
omitted.  This might be useful when in browse mode.

.TP
.B "\-\-version, \-v"
displays version and exits.

.SH NOTES
As dBase is DOS, umlauts are stored using a different code table
(namely ASCII) than most modern unices (namely ANSI).  If you encounter
such a file, I would recommend piping the output through
.BR recode (1)
with
.B ibmpc:latin1
as it's argument.

If you want to examine the output generated by the browse mode, just
take
.BR cut (1)
and set its delimiter to the used delimiter or take
.BR awk (1)
and continue.

.SH COPYRIGHT
.B Dbview
is free software. It is based on routines from unknown source that I
found on nic.funet.fi in /pub/msdos/languages/c as dbase.c.  The file
contained the following notice:

These functions are provided by Valour Software as a gift.

I have modified and included this file and wrote a skeleton around
it.  All together provides a powerful tool for dBase III and IV
database manipulation under Unix.

I mainly have written this program, because I've got several dbase
files containing important information for me.  As I won't go running
DOS everytime I need some of the stored information, I had to find a
viewer that runs unter Unix, resp. Linux, but unfortunately didn't
find one.  So it was my turn.

This package as a whole is published under the GNU Public License,
which is a great invention.

It wasn't the intention to write a freaking viewer and reinvent the
wheel again.  Instead
.B dbview
is intend to be used in conjunction with your favourite unix text
utilities like
.BR cut ,
.B recode
and
.BR more .

Martin Schulze
.br
Infodrom Oldenburg
.br
joey@infodrom.north.de
.SH "SEE ALSO"
.BR recode (1),
.BR more (1),
.BR awk (1),
.BR cut (1).


debian-bo-master/bo/source/misc/dbview-1.0.3/dbview.c
/*
    dbview.c - View dBase III files
    Copyright (c) 1995  Martin Schulze <Martin.Schulze@Infodrom.North.DE>

    This file is part of the dbview package, a viewer for dBase II files.

    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation; either version 2 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software
    Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
*/

/*
 * Thu Sep 26 21:38:33 1996:  Martin Schulze <joey@infodrom.north.de>
 *	Added trimming option (-t) to also get untrimmed browse output.
 *	Thanks to Magnus Lassus <mlassus@walli.uwasa.fi> for the idea and
 *	providing me with a patch.
 */

#include "version.h"
#include "db_dump.h"

#include <stdio.h>
#include <getopt.h>

void help_short()
{
    printf ("%s %s - %s, (c) 1996 by Martin Schulze\n", progname, version, longname);
    printf ("\n");
    printf ("%s [-b [-t]] [-d delim] [-e] [-h] [-i] [-o] [-o] [-v] dbfile\n", progname);
}

void help_long()
{
    printf ("%s %s - %s, (c) 1996 by Martin Schulze\n", progname, version, longname);
    printf ("\n");
    printf ("  --browse, -b           browse the database\n");
    printf ("  --delimiter, -d        set the delimiter for browse output\n");
    printf ("  --description, -e      display field description\n");
    printf ("  --help, -h             display help\n");
    printf ("  --info, -i             display db information\n");
    printf ("  --omit, -o             omit db records\n");
    printf ("  --reserve, -r          reserve fieldnames from beeing translated\n");
    printf ("  --trim, -t             trim browse fields\n");
    printf ("  --version, -v          display version\n");
}

void main (int argc, char **argv)
{
    int opt_index;
    char c;
    static const struct option long_options[] =
      {
	{"browse", no_argument, 0, 'b'},
	{"delimiter", required_argument, 0, 'd'},
	{"description", no_argument, 0, 'e'},
	{"help", no_argument, 0, 'H'},
	{"info", no_argument, 0, 'i'},
	{"omit", no_argument, 0, 'o'},
	{"reserve", no_argument, 0, 'r'},
	{"trim", no_argument, 0, 't'},
	{"version", no_argument, 0, 'v'},
	{0, 0, 0, 0}
	};
    char dbfile[256];
    int flags = 0;
    char delim = ':';

    optind = 0;
    while ( (c = getopt_long(argc, argv, "Hbd:ehiortv", long_options, &opt_index)) != -1 ) {
	switch (c) {
	case 'H':	/* --help */
	    help_long (); exit (0);
	case 'b':
	    flags |= DB_FL_BROWSE;
	    break;
	case 'd':
            if (!argv[optind]) {
                printf ("Too few arguments.\n");
                break;
            }
	    delim = optarg[0];
	    break;
	case 'e':
	    flags |= DB_FL_DESCR;
	    break;
	case 'h':
	    help_short (); exit (0);
	case 'i':
	    flags |= DB_FL_INFO;
	    break;
	case 'o':
	    flags |= DB_FL_OMIT;
	    break;
	case 'r':
	    flags |= DB_FL_RESERVE;
	    break;
	case 't':
	    flags |= DB_FL_TRIM;
	    break;
	case 'v':	/* --version */
	    printf ("%s %s - %s\n", progname, version, longname); exit (0);
	}   /* switch */
    }   /* while */

/*    printf ("argc=%d, optind=%d\n", argc, optind); */
    if ( argc - optind < 1) {
	help_short();
	exit (0);
    }

    strcpy(dbfile, argv[optind]);
    /* if ! -f dbfile thnen -f dbfile.dbf */

    db3_process (dbfile, flags, delim);

}
